src/fabric/test/fabric2_doc_count_tests.erl

```diff
@@ -109,6 +110,30 @@ normal_docs({Db, _}) ->
         ).
 
 
+replicated_docs({Db, _}) ->
+    {DocCount, DelDocCount, DDocCount, LDocCount} = get_doc_counts(Db),
+
+    Opts = [replicated_changes],
+    {R1, R2, R3} = {<<"r1">>, <<"r2">>, <<"r3">>},
+
+    % First case is a simple replicated update
+    Doc1 = #doc{id = <<"rd1">>, revs = {1, [R1]}},
+    {ok, {1, R1}} = fabric2_db:update_doc(Db, Doc1, Opts),
+    check_doc_counts(Db, DocCount + 1, DelDocCount, DDocCount, LDocCount),
+
+    % Here a deleted document is replicated into the db. Doc count should not
+    % change, only deleted doc count.
+    Doc2 = #doc{id = <<"rd2">>, revs = {1, [R2]}, deleted = true},
+    {ok, {1, R2}} = fabric2_db:update_doc(Db, Doc2, Opts),
+    check_doc_counts(Db, DocCount + 1, DelDocCount + 1, DDocCount, LDocCount),
+
+    % Here we extended the deleted document's rev path but keep it deleted.
+    % Again, only deleted doc count should be bumped.
```

Comment:
Pretty sure this is wrong as I mentioned above. If you have a doc that's deleted, and replicate a change that adds a revision but remains deleted, then DelDocCount should remain the same, no?

Reply:
Yap. You're right. The count shouldn't bump if it was already deleted.